### DISCUSSION TITLE
CLI / API:  assign own unique alias directly with initial deployment

### DISCUSSION THREAD

**Discussion by nkuehn** [vercel/community#1176] [Idea]
Hi, the autogenerated domains for any deployment are great as a primary setup. 

But since they have the (likely intended) issue of not being predictable we're actually discouraging their use in our setups and generally create one Alias that is based on the git SHA and a short "slug" of the project which is then used for any further steps in our workflow.  

The additional alias does not strictly hurt, but it's unnecessary load and also a bit of UI clutter to teach users to not use the vercel generated one but somehow pick our SHA based one from the hover UI element.   

It would be neat if a CLI  command to deploy could immediately assign a primary alias without an extra command and/or API call.  The vercel internal alias for the build / deploy could then either be omitted or hidden or at least be pushed down the list (I know, I can make multiple deployments from a certain SHA so it's not the same cardinality).  

Not exactly unix-ish philosophy (one command one purpose) but we did run into API limits recently due to many aliases being set and need to optimize out a few paths here.

**Comment by amyegan** [comment]
Thanks for sharing this idea @nkuehn! I want to know more about your workflow so I can share this with the right team. Are you using Vercel's GitHub integration, or are you using the CLI for deployment? I know the title says CLI / API alias, but I want to be sure of your preferred deployment method.

**Reply by nkuehn** [reply]
We moved away from building on Vercel to using the CLI from CI because it's gradually become a complex pipeline and Vercel isn't intended to be a full blown CI.  

I posted this because we do have serious issues with the API and deployment rate limits now.  We switched to deploying microfrontends in individual Vercel projects and since it's CI based all limits are attributed to the user whose token is used by CI.  So I debugged the API call sequences in the CLI and found it may be less taxing on the limits to combine deploy and setting the main git SHA based alias.  

The root cause is the limits though, there is no concept of a machine / service account user that has higher limits that cater for the situation that such a user would centrally do all the deployments.    In general, I question that many of the API limits are per-user.  Per project or even per organization may make more sense in some cases.
